### PR TITLE
Switches to Docker to run RabbitMQ integration tests

### PR DESCRIPTION
**Pull request by adriancole**
I keep forgetting to run rabbitmq-server, sometimes forgetting where it is..
This makes it work provided docker works.

### CODE REVIEW

File: instrumentation/mongodb/pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
   <parent>
     <groupId>io.zipkin.brave</groupId>
     <artifactId>brave-instrumentation-parent</artifactId>
-    <version>5.10.2-SNAPSHOT</version>
+    <version>5.10.3-SNAPSHOT</version>
```
Comment (adriancole) [inline instrumentation/mongodb/pom.xml]: master is currently broke

File: instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbitTracing.java [hunk 2]
```diff
@@ -131,4 +135,26 @@
     assertThat(consumerReporter.takeLocalSpan().name())
       .isEqualTo("on-message");
   }
+
+  @Test public void producerSampler() {
```
Comment (adriancole) [inline instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbitTracing.java]: inlining these tests saves 15s each on my laptop as the overall container bootstrap seems to have 15s overhead

File: instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbit.java [hunk 3]
```diff
@@ -37,43 +38,68 @@
...
+      amqpAdmin.declareQueue(queue);
+      amqpAdmin.declareBinding(binding);
+    } finally {
+      connectionFactory.destroy();
+    }
+  }
 
-  @ClassRule public static BrokerRunning brokerRunning = BrokerRunning.isRunning();
+  @AfterClass public static void kiwwTheWabbit() {
+    rabbit.start();
```
Comment (adriancole) [inline instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbit.java]:
```suggestion
    rabbit.stop();
```

[the remark this answers is not in the document]
Comment (adriancole) [reply instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbit.java]: typo should have been stop.. would be nice if it was named kill, so someone can get my Elmer Fudd joke

[the remark this answers is not in the document]
Comment (anuraaga) [reply instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbit.java]: The method name still conveys the joke :)

**Pull request merged**